integration_tests/src/main/python/cmp_test.py

```diff
@@ -156,6 +156,7 @@ def test_lte(data_gen):
             lambda spark : binary_op_df(spark, data_gen).select(
```

Comment:
Do we want some tests with hard coded NaN in them?

```
f.lit("NaN").cast(data_type)
```

Similar to what we do for None?  Just so we know the error case is covered instead of waiting for it to possibly happen.